core/src/avm2/globals/flash/media/sound.rs

```diff
@@ -18,8 +19,8 @@ pub fn init<'gc>(
     this: Object<'gc>,
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    if !args.is_empty() {
-        avm2_stub_constructor!(activation, "flash.media.Sound", "with arguments");
+    if args.try_get_object(activation, 0).is_some() {
+        let _ = load(activation, this, args);
```

Comment:
Just to be clear, is this intentional that it ignores error?

Reply:
I'm not really sure what Flash does here. The documentation said `If you do not pass a valid URLRequest object to the Sound constructor, you must call the load() function for the Sound object yourself, or the stream will not load.` Which would imply the Sound object still gets constructed to me. But I'm not sure.

Reply:
Unlike the `load` method, which says it throws, this does not state that it throws in the documentation, so I feel like it probably ignores error.

Reply:
I'm still not really familiar with Rust though, so this was really just a result of that. Would this be the right way to bubble the error up?
```rs
    if args.try_get_object(activation, 0).is_some() {
        let load_result = load(activation, this, args);

        let _ = match load_result {
            Ok(load) => load,
            Err(_) => return load_result,
        };
    }

```

Reply:
Closed in favor of #11972